compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -567,7 +574,22 @@ fn check_item_type(tcx: TyCtxt<'_>, id: hir::ItemId) {
         tcx.def_path_str(id.owner_id)
     );
     let _indenter = indenter();
-    match tcx.def_kind(id.owner_id) {
+    let def_kind = tcx.def_kind(id.owner_id);
+    if !matches!(def_kind, DefKind::Trait) {
+        let item_def_id = id.owner_id.to_def_id();
+        if let Some(attr) = tcx
+            .get_attrs_by_path(item_def_id, Box::new([sym::diagnostic, sym::on_unimplemented]))
+            .next()
+        {
+            tcx.emit_spanned_lint(
+                UNKNOWN_OR_MALFORMED_DIAGNOSTIC_ATTRIBUTES,
+                tcx.hir().local_def_id_to_hir_id(item_def_id.expect_local()),
+                vec![attr.span],
```

Comment:
`Span` is `Into<MultiSpan>`

```suggestion
                attr.span,
```